instaclone/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,74 @@
+import React from "react";
+import { CommentInput } from "./Styles";
+import Comment from "./Comment";
+
+class CommentSection extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      comments: this.props.comments,
+      comment: "",
+      index: this.props.index
+    };
+  }
+  addNewComment = e => {
+    if (this.state.comment) {
+      let newComments = [
+        ...this.state.comments,
+        { username: localStorage.getItem("user"), text: this.state.comment }
+      ];
+      if (e.key === "Enter") {
+        this.setState({
+          comments: newComments,
+          comment: ""
+        });
+      }
+    }
+  };
+  deleteComment = index => {
+    let filteredComments = this.state.comments.filter(
+      (comment, i) => index !== i
+    );
+    this.setState({ comments: filteredComments });
+  };
+  onCommentChange = e => {
+    this.setState({ [e.target.name]: e.target.value });
```

Comment:
an alternative using destructuring could be ==>

 ({ target }) => {
      const { name, value } = target
      this.setState({[name]: value})
    }